File: translations/codenet/uncommented/C++2C/granite/Code/s353226283.c

#include <stdio.h>
#include <stdlib.h>
#include <algorithm>
int main() {
    int n, k, t;
    int *vec;
    int ans = 1000000001;
    scanf("%d %d", &n, &k);
    vec = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &t);
        vec[i] = t;
    }
    std::sort(vec, vec + n);
    for (int i = 0; i + k -1 < n; i++) {
        int diff = vec[i+k-1] - vec[i];
        if (diff < ans) {
            ans = diff;
        }
    }
    printf("%d\n", ans);
    free(vec);
    return 0;
}

File: translations/codenet/mistral_commenetd/C++2Go/deepseek/Code/s353226283.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n, k int
	fmt.Scan(&n, &k)

	vec := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&vec[i])
	}

	sort.Ints(vec)

	ans := 1000000001
	for i := 0; i+k-1 < n; i++ {
		diff := vec[i+k-1] - vec[i]
		if diff < ans {
			ans = diff
		}
	}

	fmt.Println(ans)
}

